gpytorch/kernels/kernel.py

```diff
@@ -352,6 +352,49 @@ def covar_dist(
 
         return res
 
+    def expand(self, *sizes: Union[torch.Size, Tuple[int, ...]]) -> Kernel:
+        r"""
+        Constructs a new kernel where the lengthscale (and other kernel parameters)
+        are expanded to match the batch dimension determined by `sizes`.
+
+        :param sizes: The batch shape of the new tensor
+        """
+        # Type checking
+        if len(sizes) == 1 and hasattr(sizes, "__iter__"):
+            new_batch_shape = torch.Size([*(sizes[0])])
```

Comment:
Does this not work? If you're checking the `len` then this wouldn't work with generators anyway, no? Or is this some hack around type checking weirdness?
```suggestion
            new_batch_shape = torch.Size(sizes[0])
```